util/src/main/java/com/redhat/lightblue/util/statsd/StatsdMetricObserver.java

```diff
@@ -0,0 +1,72 @@
+package com.redhat.lightblue.util.statsd;
+
+import java.util.List;
+
+import com.netflix.servo.Metric;
+import com.netflix.servo.publish.BaseMetricObserver;
+import com.netflix.servo.publish.graphite.BasicGraphiteNamingConvention;
+import com.netflix.servo.publish.graphite.GraphiteNamingConvention;
+import com.timgroup.statsd.NonBlockingStatsDClient;
+import com.timgroup.statsd.StatsDClient;
+import com.timgroup.statsd.StatsDClientErrorHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Metrics observer that sends hystrix data to statsd with the intent of then then
+ * sending it to graphite.
+ *
+ * @author nmalik, dcrissman
+ */
+public class StatsdMetricObserver extends BaseMetricObserver {
+    private static final Logger LOGGER = LoggerFactory.getLogger(StatsdMetricObserver.class);
+
+    private final String prefix;
+    private final String host;
+    private final int port;
+    private final GraphiteNamingConvention namingConvention;
+
+    public StatsdMetricObserver(String prefix, String host, int port) {
+        this(prefix, host, port, new BasicGraphiteNamingConvention());
+    }
+
+    public StatsdMetricObserver(String prefix, String host, int port, GraphiteNamingConvention namingConvention) {
+        super("StatsdMetricObserver." + prefix);
+        this.prefix = prefix;
+        this.host = host;
+        this.port = port;
+        this.namingConvention = namingConvention;
+    }
+
+    @Override
+    public void updateImpl(List<Metric> metrics) {
+        // The statsd client doesn't do any checks on the underlying socket's state
+        // and the socket connects only once, so we cannot trust the socket to stay
+        // open over a period of time.  If this is changed/fixed we could reuse the
+        // client but until then it cannot be safely reused.
+        StatsDClient statsd = new NonBlockingStatsDClient(prefix, host, port, new ErrorHandler());
+        LOGGER.debug("sending data");
+
+        for (Metric metric : metrics) {
+            String aspect = namingConvention.getName(metric);
+
+            // TODO figure out how to determine if this is a guage vs counter
```

Comment:
Looking the source code of servo, I would say that the metrics that is gauge  (extending com/netflix/servo/monitor/Gauge.java ) would have a tag DataSourceType.GAUGE ( com/netflix/servo/annotations/DataSourceType.java) on it . You could also check for the DataSourceType.COUNTER tag as well